### ISSUE TITLE
(🐞) false error `unnecessary-ellipsis` when ellipsis (`...`) is used as a value

### ISSUE THREAD

**Issue by KotlinIsland** [PyCQA/pylint#5973]
### Bug description

```py
from types import EllipsisType


def foo(a: EllipsisType = ...) -> None:
    ""
```

Seems to be that the doc string triggers it for some reason.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:4:26: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.13.0
astroid 2.11.1
Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
```
